move/squid/sources/squid/deepbook_v3.move

```diff
@@ -0,0 +1,215 @@
+module squid::deepbook_v3 {
+    use std::type_name;
+    use std::ascii::{Self, String};
+
+    use sui::clock::Clock;
+    use sui::bcs::{Self, BCS};
+
+    use deepbook::pool::Pool;
+    use token::deep::DEEP;
+
+    use axelar_gateway::discovery::{Self, MoveCall};
+
+    use squid::swap_info::{SwapInfo};
+    use squid::squid::Squid;
+
+    const EWrongSwapType: u64 = 0;
+    const EWrongPool: u64 = 1;
+    const EWrongCoinType: u64 = 2;
+
+    const SWAP_TYPE: u8 = 1;
+
+    public struct DeepbookV3SwapData has drop {
+        swap_type: u8,
+        pool_id: address,
+        has_base: bool,
+        min_output: u64,
+        base_type: String,
+        quote_type: String,
+        lot_size: u64,
+        should_sweep: bool,
+    }
+
+    /// Estimate the output of a swap. If the output is less than the minimum output, the swap is skipped.
+    /// If the swap is not skipped, the estimate is stored in the coin bag.
+    public fun estimate<B, Q>(self: &mut SwapInfo, pool: &Pool<B, Q>, clock: &Clock) {
+        let data = self.get_data_swapping();
+        if (data.length() == 0) return;
+        let swap_data = peel_swap_data(data);
+
+        assert!(swap_data.swap_type == SWAP_TYPE, EWrongSwapType);
+        assert!(swap_data.pool_id == object::id_address(pool), EWrongPool);
+        assert!(
+            &swap_data.base_type == &type_name::get<B>().into_string(),
+            EWrongCoinType,
+        );
+        assert!(
+            &swap_data.quote_type == &type_name::get<Q>().into_string(),
+            EWrongCoinType,
+        );
+
+        if (swap_data.has_base) {
+            let (base_out, quote_out, _deep_required) = pool.get_quote_quantity_out(
+                self.coin_bag().get_estimate<B>(),
```

Comment:
Should we reduce the input here by 10 bps? Since we will reduce the input amount by 10 bps later on.